### PR TITLE
Remove `CollectionEntry`

### LINKED ISSUE #774: Remove `CollectionEntry` in favor of native & logical types
After adding native & logical types into the Flow, we should remove `CollectionEntry` as obsolete. New types that cover it better are: `ListType`, `MapType` & `StructureType`.

### PR DESCRIPTION
**Pull request by stloyd**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    <li>Remove `CollectionEntry`</li>
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

Closes #774

### CODE REVIEW

File: src/core/etl/tests/Flow/ETL/Tests/Unit/DataFrameTest.php [hunk 1]
```diff
@@ -355,12 +355,6 @@ public function extract(FlowContext $context) : \Generator
                                     new IntegerEntry('item-id', 1),
                                     new StringEntry('name', 'one'),
                                 ),
-                                new Row\Entry\CollectionEntry(
-                                    'tags',
-                                    new Row\Entries(new IntegerEntry('item-id', 1), new StringEntry('name', 'one')),
-                                    new Row\Entries(new IntegerEntry('item-id', 2), new StringEntry('name', 'two')),
-                                    new Row\Entries(new IntegerEntry('item-id', 3), new StringEntry('name', 'three'))
-                                ),
```
Comment (norberttech) [inline src/core/etl/tests/Flow/ETL/Tests/Unit/DataFrameTest.php:366]: This should be replaced MapType or StructType

File: src/core/etl/tests/Flow/ETL/Tests/Unit/Row/Schema/Constraint/NotEmptyTest.php [hunk 2]
```diff
@@ -29,7 +27,6 @@ public function test_not_empty_is_satisfied() : void
 
         $this->assertTrue($constraint->isSatisfiedBy(Entry::array('e', [1])));
         $this->assertTrue($constraint->isSatisfiedBy(Entry::boolean('e', false)));
-        $this->assertTrue($constraint->isSatisfiedBy(Entry::collection('e', new Entries())));
```
Comment (norberttech) [inline src/core/etl/tests/Flow/ETL/Tests/Unit/Row/Schema/Constraint/NotEmptyTest.php:34]: this should be replaced with map first

File: src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php [hunk 3]
```diff
@@ -72,12 +72,6 @@ public function test_getting_schema_from_row() : void
                 Entry::integer('item-id', 1),
                 Entry::string('name', 'one'),
             ),
-            Entry::collection(
-                'tags',
-                new Row\Entries(Entry::integer('item-id', 1), Entry::string('name', 'one')),
-                new Row\Entries(Entry::integer('item-id', 2), Entry::string('name', 'two')),
-                new Row\Entries(Entry::integer('item-id', 3), Entry::string('name', 'three'))
-            ),
```
Comment (norberttech) [inline src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php:129]: just like above, this should be replaced with another type, ideally Map or Struct

File: src/core/etl/tests/Flow/ETL/Tests/Unit/Row/Schema/Constraint/NotEmptyTest.php (hunk 2 above)
Comment (stloyd) [reply src/core/etl/tests/Flow/ETL/Tests/Unit/Row/Schema/Constraint/NotEmptyTest.php:34]: We don't have map yet, but the structure is already covered in the test (at end).

File: src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php (hunk 3 above)
Comment (stloyd) [reply src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php:129]: We don't have map yet, but the structure is already covered in the test (above this one).

File: src/core/etl/tests/Flow/ETL/Tests/Unit/DataFrameTest.php (hunk 1 above)
Comment (stloyd) [reply src/core/etl/tests/Flow/ETL/Tests/Unit/DataFrameTest.php:366]: We don't have map yet, but the structure is already covered in the test (above).

File: src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php (hunk 3 above)
Comment (norberttech) [reply src/core/etl/tests/Flow/ETL/Tests/Unit/RowTest.php:129]: yup, that's why it can't be merged now

**Review by norberttech — APPROVED** [review]

**Pull request merged**